parent/pom.xml

```diff
@@ -1948,156 +1945,165 @@
                   <artifactId>maven-antrun-plugin</artifactId>
                   <executions>
                      <execution>
-                        <phase>install</phase>
+                        <id>delete-existing-jacoco-report</id>
+                        <phase>initialize</phase>
                         <goals>
                            <goal>run</goal>
                         </goals>
-                        <configuration combine.self="override">
+                        <configuration>
                            <target>
-
-                              <taskdef classname="org.jacoco.ant.ReportTask" name="report">
-                                 <classpath path="${basedir}/target/jacoco-jars/org.jacoco.ant.jar" />
-                                 <classpath path="target/jacoco-jars/org.jacoco.core.jar" />
-                                 <classpath path="target/jacoco-jars/org.jacoco.report.jar" />
-                                 <classpath path="target/jacoco-jars/asm-all.jar" />
-                              </taskdef>
-
-                              <echo message="Deleting previous report" />
-                              <delete dir="${dir.report}" />
-
-                              <echo message="Creating report directory" />
-                              <mkdir dir="${dir.report}" />
-
-                              <echo message="Generating JaCoCo Reports" />
-                              <report>
-
-                                 <executiondata>
-                                    <fileset dir="${dir.ispn}" includes="**/target/jacoco.exec" />
-                                 </executiondata>
-
-
-                                 <structure name="${reportTitle}">
-                                    <classfiles>
-                                       <fileset dir="${dir.ispn}">
-                                          <include name="**/target/classes/" />
-                                       </fileset>
-                                    </classfiles>
-                                    <sourcefiles encoding="UTF-8">
-                                       <mappedresources>
-                                          <fileset dir="${dir.ispn}" includes="**/src/main/java/**/*.java" />
-                                          <mapper from="^(.*/java/)(.*)$" to="\2" type="regexp" />
-                                       </mappedresources>
-                                    </sourcefiles>
-
-                                 </structure>
-
-
-                                 <html destdir="${dir.report}" />
-                              </report>
+                              <echo>Deleting existing Jacoco Report Directory.</echo>
+                              <delete dir="${dir.jacoco.report}"/>
+                              <delete dir="${dir.jacoco.merged}"/>
+                              <delete dir="${project.build.directory}/classes" />
```

Comment:
This line removes the outdated classes/ folder from infinispan/parent folder. Originally it doesn't contain classes/ folder. I am adding manually the classes from all modules there for proper jacoco code coverage report generation.